File: apps/mobile-ios/Models/Conversation.swift
import Foundation

// MARK: - Conversation Model
struct Conversation: Identifiable, Codable {
    let id: String
    let title: String
    var messages: [Message]
    let createdAt: Date
    var updatedAt: Date
    var isActive: Bool
    var context: ConversationContext?

    init(
        id: String = UUID().uuidString,
        title: String,
        messages: [Message] = [],
        createdAt: Date = Date(),
        updatedAt: Date = Date(),
        isActive: Bool = true,
        context: ConversationContext? = nil
    ) {
        self.id = id
        self.title = title
        self.messages = messages
        self.createdAt = createdAt
        self.updatedAt = updatedAt
        self.isActive = isActive
        self.context = context
    }
}

// MARK: - Conversation Context
struct ConversationContext: Codable {
    let userId: String
    var preferences: [String: String]
    var metadata: [String: String]
}

// MARK: - Conversation Extensions
extension Conversation {
    var lastMessage: Message? {
        messages.last
    }

    var lastMessagePreview: String {
        lastMessage?.content ?? "No messages"
    }

    var unreadCount: Int {
        messages.filter { $0.status != .read && $0.role == .assistant }.count
    }
}

File: apps/mobile-android/app/src/main/kotlin/ai/tide/app/data/models/Conversation.kt
package ai.tide.app.data.models

import java.util.UUID
import java.util.Date

// MARK: - Conversation Model
data class Conversation(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val messages: MutableList<Message> = mutableListOf(),
    val createdAt: Date = Date(),
    var updatedAt: Date = Date(),
    var isActive: Boolean = true,
    val context: ConversationContext? = null
) {
    val lastMessage: Message?
        get() = messages.lastOrNull()

    val lastMessagePreview: String
        get() = lastMessage?.content ?: "No messages"

    val unreadCount: Int
        get() = messages.count { it.status != MessageStatus.READ && it.role == MessageRole.ASSISTANT }
}

// MARK: - Conversation Context
data class ConversationContext(
    val userId: String,
    val preferences: Map<String, String> = emptyMap(),
    val metadata: Map<String, String> = emptyMap()
)
